### PR TITLE
docs: add Clay code plugin documentation

### PR DESCRIPTION
**Pull request by mynaramana**
Clay code plugin documentation and fixing the copyright on the clay code source files.

Author: Myna
Signed-off-by: Myna <mynaramana@gmail.com>

- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

**Comment by tchaikov** [comment]
jenkins render docs

**Comment by tchaikov** [comment]
could you prefix the title of your commit message with the subcomponent your are changing ? see https://github.com/ceph/ceph/blob/master/SubmittingPatches.rst#3-describe-your-changes 

also, the "Author" field in your commit message is redundant, as it's implied in the commit itself.

File: doc/rados/operations/erasure-code-clay.rst [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+from an available OSD is only a fraction of a chunk. In fact the number
+of sub-chunks within a chunk that are accessed during repair is given by::
+
+	repair sub-chunk count = sub chunk count / q
+
+Examples 
+--------
+
+1. For a configuration with *k=4*, *m=2*, *d=5*, the sub-chunk count is
+8 and  the repair sub-chunk count is 4. Therefore, only half of a chunk is read
```
Comment (tchaikov) [inline doc/rados/operations/erasure-code-clay.rst]: extra space.

File: doc/rados/operations/erasure-code-clay.rst [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+              the number of OSDs that can be down without losing data.
+
+:Type: Integer
+:Required: Yes.
+:Example: 2
+
+``d={helper-chunks}``
+
+:Description: Number of OSDs requested to send data during recovery of
+              a single chunk.
```
Comment (tchaikov) [inline doc/rados/operations/erasure-code-clay.rst]: might want 

1. note the range of possible `d`
1. explain the relation between amount data to transmit for recovery and the choosing of `d`, people might want to use minimal possible `d`. as it sounds like the less number of OSDs to contact the better, at first glance.

File: doc/rados/operations/erasure-code-clay.rst [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+2. When *k=8*, *m=4*, *d=11* the sub-chunk count is 64 and repair sub-chunk count
+is 16. A quarter of a chunk is read from an available OSD for repair of a failed 
+chunk.
+
+
+
+How to choose configuration given a workload
+============================================
+Only few sub-chunks are read of all the subchunks within a chunk. These
+sub-chunks need not be contiguously placed within a chunk. For best disk IO
```
Comment (tchaikov) [inline doc/rados/operations/erasure-code-clay.rst]: could rephrase this sentence like

> These sub-chunks are not necessarily stored consecutively within a chunk.

**Comment by tchaikov** [comment]
jenkins render docs

**Comment by mynaramana** [comment]
jenkins render docs

**Comment by tchaikov** [comment]
@mynaramana the indent in reST matters. i posted https://github.com/mynaramana/ceph/pull/1 to address some of the indent issues

you can use 

```sh
cd admin
./build-doc html
./serve-doc 
```
for building and viewing the built doc at `http://localhost:8080`.

**Comment by tchaikov** [comment]
jenkins render docs

**Comment by tchaikov** [comment]
regarding to b8a752198e434ddf68e34cd327327a22df024293 
could you prefix the title of your commit message with the subcomponent your are changing ? see https://github.com/ceph/ceph/blob/master/SubmittingPatches.rst#3-describe-your-changes , like 

```
erasure-code: fixes the copyright of clay source and cleanups
```

**Comment by mynaramana** [comment]
retest this please

**Pull request merged**